zerver/webhooks/insping/view.py

```diff
@@ -0,0 +1,44 @@
+from django.utils.translation import ugettext as _
+from zerver.lib.actions import check_send_stream_message
+from zerver.lib.response import json_success, json_error
+from zerver.decorator import REQ, has_request_variables, \
+    api_key_only_webhook_view
+from zerver.lib.validator import check_dict, check_string
+
+from zerver.models import Client, UserProfile
+
+from django.http import HttpRequest, HttpResponse
+from typing import Dict, Any, Iterable, Optional, Text
+
+import time
+
+
+@api_key_only_webhook_view('Insping')
+@has_request_variables
+def api_insping_webhook(
+        request: HttpRequest, user_profile: UserProfile,
+        payload: Dict[str, Dict[str, Any]]=REQ(argument_type='body'),
+        stream: Text=REQ(default='test'),
+        topic: Text=REQ(default='insping')
+) -> HttpResponse:
+
+    data = payload['webhook_event_data']
+
+    state_name = data.get('check_state_name')
+    url_tested = data.get('request_url')
+    response_time = data.get('response_time')
+    timestamp = data.get('request_start_time')
```

Comment:
I wouldn't use `dict.get` here. I would recommend using `data['request_start_time']`. `dict.get` can return `None`. If a payload should always have something, it should be able to complain if in certain instances, it doesn't. :)

Reply:
Insping payload always have those things, so I'll change it into the suggestion instead, thanks!